test/unit/circuits/test_qaoa.py

```diff
@@ -0,0 +1,105 @@
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test QAOA circuits module."""
+
+
+from pytest import mark, raises  # pylint: disable=import-error
+from qiskit import QuantumCircuit
+
+from quantum_enablement.circuits._qaoa import QAOALineCircuit
+
+
+def two_qubit_depth_filter(instruction):
+    """Filter instructions which do not contribute to two-qubit depth."""
+    num_qubits = instruction.operation.num_qubits
+    name = instruction.operation.name
+    return num_qubits > 1 and name != "barrier"
+
+
+class TestQAOALineCircuit:
+    """Test suite for the QAOALineCircuit class."""
+
+    @mark.parametrize("num_qubits", [4, 6])
+    @mark.parametrize("depth", [0, 2, 4])
+    def test_creation(self, num_qubits, depth):
+        """Test QAOALineCircuit creation."""
+        circuit = QAOALineCircuit(num_qubits, depth, barriers=False, measurements=False)
+        op_counts = circuit.count_ops()
+        assert isinstance(circuit, QuantumCircuit), "QAOALineCircuit must be a QuantumCircuit."
+        assert circuit.num_qubits == num_qubits, "Wrong number of qubits."
+        assert circuit.depth(two_qubit_depth_filter) == depth, "Wrong two-qubit depth."
+        assert circuit.num_parameters == depth, "Wrong number of parameters."
+        assert op_counts.get("h", 0) == num_qubits, "Wrong number of H gates."
+        assert op_counts.get("rzz", 0) == (num_qubits - 1) * (
+            depth // 2
+        ), "Wrong number of RZZ gates."
+        assert op_counts.get("rx", 0) == num_qubits * (depth // 2), "Wrong number of RX gates."
+        assert op_counts.get("barrier", 0) == 0, "Wrong number of barriers."
+        assert op_counts.get("measure", 0) == 0, "Wrong number of measurements."
+
+    def test_defaults(self):
+        """Test QAOALineCircuit defaults."""
+        circuit = QAOALineCircuit(4, 4)  # Note: default barriers and measurements are False.
+        op_counts = circuit.count_ops()
+        assert op_counts.get("barrier", 0) == 0, "By default barriers should not be present."
+        assert op_counts.get("measure", 0) == 0, "By default measurements should not be present."
+
+    @mark.parametrize("num_qubits", [4, 6])
+    @mark.parametrize("depth", [0, 2, 4])
+    def test_barriers(self, num_qubits, depth):
+        """Test QAOALineCircuit barriers."""
+        circuit = QAOALineCircuit(num_qubits, depth, barriers=True, measurements=False)
+        op_counts = circuit.count_ops()
+        assert op_counts.get("barrier", 0) == depth // 2, "Wrong number of barriers."
+        # circuit.remove_operations("barrier")
+        # assert circuit == QAOALineCircuit(num_qubits, depth, measurements=False, barriers=False)
```

Comment:
These comments should be removed.

Reply:
Good point @a-matsuo ! Your approach is definitely the standard way to go and what we should pursue.

However, those comments are there because the `remove_barriers` method does not exist in Qiskit but would make the tests more complete; so I would like to leave them there for future reference and possible contribution to Qiskit.